File: node/rate_calculator/rate_calculator.cpp
#include <vector>
#include <iostream>

double calculate_rate(const std::vector<double>& contracts) {
    if (contracts.empty()) {
        std::cerr << "Error: The contracts vector is empty." << std::endl;
        return 0.0;
    }

    double sum = 0.0;
    for (double value : contracts) {
        sum += value;
    }
    return sum / contracts.size();
}

File: node/rate_calculator/rate_calculator.js
function calculateRate(contracts) {
  if (contracts.length === 0) {
    console.error("Error: The contracts array is empty.");
    return 0.0;
  }

  let sum = contracts.reduce((acc, value) => acc + value, 0);
  return sum / contracts.length;
}
module.exports = calculateRate;